core/include/esp/iomux.h

```diff
@@ -43,10 +43,17 @@ inline static esp_reg_t gpio_iomux_reg(const uint8_t gpio_number)
     return &(IOMUX.PIN[gpio_to_iomux(gpio_number)]);
 }
 
+/**
+ * Set the I/O Mux function. The iomux_num is an IOMUX register number, see
+ * gpio_iomux_reg above to obtain the IOMUX register number of a GPIO number.
+ * The func is an IOMUX_GPIO<n>_FUNC_<function> value, see iomux_regs.h
+ *
+ * ie *gpio_iomux_reg(3) is equivalent to IOMUX_GPIO3
```

Comment:
You can't pass IOMUX_GPIO3 into this function as *iomux_num* parameter. *iomux_num* is an index of a register. IOMUX_GPIO3 is a register itself.
It is still a confusing function with confusing parameters.